packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -296,6 +296,63 @@ class UiKitView extends StatefulWidget {
   State<UiKitView> createState() => _UiKitViewState();
 }
 
+/// Embeds arbitrary HTML in the Widget hierarchy.
+///
+/// Embedding HTML is an expensive operation and should be avoided when a
+/// Flutter equivalent is possible.
+///
+/// The embedded HTML is painted just like any other Flutter widget and
+/// transformations apply to it as well. This widget should only be used in
+/// Flutter Web.
+///
+/// {@macro flutter.widgets.platformViews.layout}
+///
+/// Due to security restrictions with cross-origin `<iframe>` elements, Flutter
+/// cannot dispatch pointer events to an HTML view. If an `<iframe>` is the
+/// target of an event, the window containing the `<iframe>` is not notified
+/// of the event. In particular, this means that any pointer events which land
+/// on an `<iframe>` will not be seen by Flutter, and so the HTML view cannot
+/// participate in gesture detection with other widgets.
+///
+/// The way we enable accessibility on Flutter for web is to have a full-page
+/// button which waits for a double tap. Placing this full-page button in front
+/// of the scene would cause platform views not to receive pointer events. The
+/// tradeoff is that by placing the scene in front of the semantics placeholder
+/// will cause platform views to block pointer events from reaching the
+/// placeholder. This means that in order to enable accessibility, you must
+/// double tap the app *outside of a platform view*. As a consequence, a
+/// full-screen platform view will make it impossible to enable accessibility.
+/// Make sure that your HTML views are sized no larger than necessary, or you
+/// may cause difficulty for users trying to enable accessibility.
+///
+/// {@macro flutter.widgets.platformViews.lifetime}
+class HtmlView extends StatefulWidget {
+  /// Creates a widget that embeds HTML.
+  ///
+  /// {@macro flutter.widgets.platformViews.constructorParams}
+  const HtmlView({
+    Key key,
+    @required this.viewType,
+    this.onPlatformViewCreated,
+    this.layoutDirection,
+  }) : assert(viewType != null),
+       super(key: key);
+
+  /// The unique identifier for the HTML view type to be embedded by this widget.
+  ///
+  /// A PlatformViewFactory for this type must have been registered.
```

Comment:
Says, that PlatformViewFactory must be registered, but how??

Reply:
There's an example in the flutter_web fork here https://github.com/flutter/flutter_web/blob/master/examples/html_platform_view/lib/main.dart and it looks like the corresponding code in the main branch is in the engine at https://github.com/flutter/engine/blob/master/lib/web_ui/lib/src/engine/platform_views.dart

Which you would have to register in a plugin? An example of which for ios is in the webview_flutter plugin here https://github.com/flutter/plugins/blob/560694c5b58eed8a46e57b1dd61e701591209340/packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewFlutterPlugin.java

Not sure what this plugin registration would look like on the web platform though?